editor/src/clj/editor/engine/native_extensions.clj

```diff
@@ -152,9 +152,10 @@
    (g/with-auto-evaluation-context evaluation-context
      (get-build-server-url prefs project evaluation-context)))
   (^String [prefs project evaluation-context]
-   (or (not-empty (prefs/get-prefs prefs "extensions-server" ""))
-       (not-empty (shared-editor-settings/get-setting project ["extensions" "build_server"] evaluation-context))
-       defold-build-server-url)))
+   (string/trim
+     (or (not-empty (prefs/get-prefs prefs "extensions-server" ""))
+         (not-empty (shared-editor-settings/get-setting project ["extensions" "build_server"] evaluation-context))
+         defold-build-server-url))))
```

Comment:
Hmm now that got me thinking... If the user enters a build server url in preferences to be a non-empty string only with spaces, it will be trimmed and "used". We could prevent this error from happening if we trimmed the setting value before checking that it's not empty. We also don't need to trim `defold-build-server-url` since it's set by us to a string that is known to be trimmed... Then, the code would look like that:
```clj
(or (not-empty (string/trim (prefs/get-prefs prefs "extensions-server" ""))) ;; always trim because `get-prefs` does not return nil
    (not-empty (some-> (shared-editor-settings/get-setting project ["extensions" "build_server"] evaluation-context) string/trim)) ;; use `some->` because `get-setting` may return nil
    defold-build-server-url)
```

Reply:
I kept your comments in code